library/core/src/ops/deref.rs

```diff
@@ -309,3 +309,25 @@ impl<T: ?Sized> Receiver for &T {}
 
 #[unstable(feature = "receiver_trait", issue = "none")]
 impl<T: ?Sized> Receiver for &mut T {}
+
+#[lang = "stable_deref"]
```

Comment:
This shouldn't be made into a lang item until it's actually used by the compiler. I feel like having unused lang items is not desirable.

Reply:
is there any chance of us somehow managing to check and error in that case?

Reply:
*thinks* okay the moment this left my hands the answer came to me as "probably not usefully".

Reply:
I think the use of it is rather imminent. Ideally with #123472 this would go into the trait bounds that `#[derive(SmartPointer)]` generates.

Reply:
I came to my right sense! I have dropped the lang term here.